apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -928,6 +930,21 @@ public async Task ShutdownPoolAsync_CannotAcquireAfterShutdown()
                 });
             }
 
+            [Fact]
+            public void BatchCreateSession_WithDatabaseRole()
+            {
+                Mock<SpannerClient> spannerClientMock = SpannerClientHelpers.

```

Comment:
Instead of using a mock here, can't you use the same (with maybe some slight modifications) as we are using for the rest of all other tests?